Source/PlayerEx.cpp

```diff
@@ -101,25 +116,103 @@ int PlayerEx::GetSatelliteCount(map_id objectType)
 		return playerArray[playerNum].satellites.solarSat;
 	case mapEDWARDSatellite:
 		return playerArray[playerNum].satellites.edwardSat;
+	case mapEvacuationModule:
+		return playerArray[playerNum].satellites.evacModule;
+	case mapChildrenModule:
+		return playerArray[playerNum].satellites.childCargo;
+	case mapRareMetalsCargo:
+		return playerArray[playerNum].satellites.rareCargo;
+	case mapCommonMetalsCargo:
+		return playerArray[playerNum].satellites.commonCargo;
+	case mapFoodCargo:
+		return playerArray[playerNum].satellites.foodCargo;
+	case mapPhoenixModule:
+		return playerArray[playerNum].satellites.phoenixModule;
+	case mapOrbitalPackage:
+		return playerArray[playerNum].satellites.orbitalPackage;
+	case mapStasisSystems:
+		return playerArray[playerNum].satellites.stasisSystems;
+	case mapSkydock:
+		return playerArray[playerNum].satellites.skydock;
+	case mapSensorPackage:
+		return playerArray[playerNum].satellites.sensorPackage;
+	case mapHabitatRing:
+		return playerArray[playerNum].satellites.habitatRing;
+	case mapFuelingSystems:
+		return playerArray[playerNum].satellites.fuelingSystems;
+	case mapCommandModule:
+		return playerArray[playerNum].satellites.commandModule;
+	case mapFusionDriveModule:
+		return playerArray[playerNum].satellites.fusionDrive;
+	case mapIonDriveModule:
+		return playerArray[playerNum].satellites.ionDrive;
 	default:
 		return 0;
 	}
 }
 
 void PlayerEx::SetSatelliteCount(map_id objectType, int count)
 {
-	if (!isInited) {
+	if (!isInited)
 		return;
-	}
 
 	switch (objectType)
 	{
 	case mapRLV:
+		// Note: changing the player's RLV count doesn't seem to actually give that player RLVs - I'm guessing there's some kind of "landing timer" that needs to be set, otherwise they remain in orbit forever.
 		playerArray[playerNum].satellites.rlv = count;
+		break;
 	case mapSolarSatellite:
 		playerArray[playerNum].satellites.solarSat = count;
+		break;
 	case mapEDWARDSatellite:
 		playerArray[playerNum].satellites.edwardSat = count;
+		break;
+	case mapEvacuationModule:
+		playerArray[playerNum].satellites.evacModule = 1;
+		break;

```

Comment:
This may be unexpected if the `count` parameter was set to `0`.

Reply:
I added a check to normalize the count to 0 or 1 for starship modules (since other values cause unexpected behavior).